jetbrains-core/src/software/aws/toolkits/jetbrains/services/sqs/toolwindow/SqsWindow.kt

```diff
@@ -3,76 +3,71 @@
-    private val toolWindow = ToolkitToolWindowManager.getInstance(project, SQS_TOOL_WINDOW)
-    private val edtContext = getCoroutineUiContext()
+class SqsWindow(private val project: Project) {
     private val client: SqsClient = project.awsClient()
 
     fun pollMessage(queue: Queue) {
-        showQueue(queue, SqsWindowUi(project, client, queue).apply { pollMessage() })
+        showQueue(queue).pollMessage()
     }
 
     fun sendMessage(queue: Queue) {
-        showQueue(queue, SqsWindowUi(project, client, queue).apply { sendMessage() })
+        showQueue(queue).sendMessage()
     }
 
-    private fun showQueue(queue: Queue, component: SqsWindowUi) = launch {
+    private fun showQueue(queue: Queue): SqsWindowUi {
         SqsTelemetry.openQueue(project, queue.telemetryType())
-        try {
-            val existingWindow = toolWindow.find(queue.queueUrl)
-            if (existingWindow != null) {
-                withContext(edtContext) {
-                    // We dispose the existing one so we can switch tabs, currently we have no other way to do this
-                    // TODO fix this to switch tabs instead of disposing when we can get the original component out of
-                    // toolwindow
-                    existingWindow.dispose()
-                }
-            }
-
-            withContext(edtContext) {
-                toolWindow.addTab(queue.queueName, component.mainPanel, activate = true, id = queue.queueUrl)
-            }
-        } catch (e: Exception) {
-            LOG.error(e) { "Exception thrown while trying to open queue '${queue.queueName}'" }
-            throw e
+
+        val toolWindow = SqsWindowFactory.getToolWindow(project)
+        val contentManager = toolWindow.contentManager
+
+        val sqsViewContent = findQueue(queue.queueUrl) ?: createSqsView(contentManager, queue)
+
+        toolWindow.show()
+        contentManager.setSelectedContent(sqsViewContent, true)
+
+        return sqsViewContent.component as SqsWindowUi
+    }
+
+    fun findQueue(queueUrl: String): Content? {
+        val toolWindow = SqsWindowFactory.getToolWindow(project)
+        val contentManager = toolWindow.contentManager
+
+        return contentManager.contents.find {
+            val component = it.component
+            component is SqsWindowUi && component.queue.queueUrl == queueUrl
         }
     }
 
-    fun closeQueue(queueUrl: String) = runBlocking {
-        withContext(edtContext) {
-            toolWindow.find(queueUrl)?.dispose()
+    private fun createSqsView(contentManager: ContentManager, queue: Queue): Content {
+        val sqsView = SqsWindowUi(project, client, queue)
+        val sqsViewContent = contentManager.factory.createContent(sqsView.component, queue.queueName, false)
+        sqsViewContent.isCloseable = true
+        sqsViewContent.isPinnable = true
+        contentManager.addContent(sqsViewContent)
```

Comment:
addContent definitely has an EDT assertion

Reply:
Ya, all our call sites our on EDT, but will add asserts to force that